Mathlib/AlgebraicGeometry/GammaSpecAdjunction.lean

```diff
@@ -0,0 +1,530 @@
+    (NatIso.ofComponents (fun X => Iso.refl _) )
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.Γ_Spec.adjunction AlgebraicGeometry.ΓSpec.adjunction
+
+theorem adjunction_homEquiv_apply {X : Scheme} {R : CommRingCatᵒᵖ}
+    (f : (op <| Scheme.Γ.obj <| op X) ⟶ R) :
+    ΓSpec.adjunction.homEquiv X R f = locallyRingedSpaceAdjunction.homEquiv X.1 R f := by
+  dsimp [adjunction, Adjunction.restrictFullyFaithful]
+  simp only [Category.comp_id, Category.id_comp]
+  rfl --Porting Note: Added
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.Γ_Spec.adjunction_hom_equiv_apply AlgebraicGeometry.ΓSpec.adjunction_homEquiv_apply
+
+theorem adjunction_homEquiv (X : Scheme) (R : CommRingCatᵒᵖ) :
+    ΓSpec.adjunction.homEquiv X R = locallyRingedSpaceAdjunction.homEquiv X.1 R :=
+  Equiv.ext fun f => adjunction_homEquiv_apply f
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.Γ_Spec.adjunction_hom_equiv AlgebraicGeometry.ΓSpec.adjunction_homEquiv
+
+theorem adjunction_homEquiv_symm_apply {X : Scheme} {R : CommRingCatᵒᵖ}
+    (f : X ⟶ Scheme.Spec.obj R) :
+    (ΓSpec.adjunction.homEquiv X R).symm f = (locallyRingedSpaceAdjunction.homEquiv X.1 R).symm f :=
+  by rw [adjunction_homEquiv]; rfl
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.Γ_Spec.adjunction_hom_equiv_symm_apply AlgebraicGeometry.ΓSpec.adjunction_homEquiv_symm_apply
+
+@[simp]
+theorem adjunction_counit_app {R : CommRingCatᵒᵖ} :
+    ΓSpec.adjunction.counit.app R = locallyRingedSpaceAdjunction.counit.app R := by
+  rw [← Adjunction.homEquiv_symm_id, ← Adjunction.homEquiv_symm_id,
+    adjunction_homEquiv_symm_apply]
+  rfl
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.Γ_Spec.adjunction_counit_app AlgebraicGeometry.ΓSpec.adjunction_counit_app
+
+@[simp]
+theorem adjunction_unit_app {X : Scheme} :
+    ΓSpec.adjunction.unit.app X = locallyRingedSpaceAdjunction.unit.app X.1 := by
+  rw [← Adjunction.homEquiv_id, ← Adjunction.homEquiv_id, adjunction_homEquiv_apply]; rfl
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.Γ_Spec.adjunction_unit_app AlgebraicGeometry.ΓSpec.adjunction_unit_app
+
+-- Porting Note: Commented
+--attribute [local semireducible] locallyRingedSpaceAdjunction ΓSpec.adjunction
+
+-- Porting Note: Had to increase Heartbeats
+set_option maxHeartbeats 250000 in
+instance isIso_locallyRingedSpaceAdjunction_counit : IsIso locallyRingedSpaceAdjunction.counit :=
+  IsIso.of_iso_inv (NatIso.op SpecΓIdentity) -- Porting Note: Had to make this explicit
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.Γ_Spec.is_iso_LocallyRingedSpace_adjunction_counit AlgebraicGeometry.ΓSpec.isIso_locallyRingedSpaceAdjunction_counit
+
+instance isIso_adjunction_counit : IsIso ΓSpec.adjunction.counit := by
+  -- Porting Note: Next line was
+  -- `apply (config := { instances := false }) nat_iso.is_iso_of_is_iso_app`
```

Comment:
This is allowed now, but renamed to `allowSynthFailures := false`. It may not work until we bump the Lean4 version tomorrow, however.

Reply:
amended